Doesn't compile with musl [API-1026]

Compiling on a linux system with musl libc (e.g. Alpine) produce the following:

```
/data/hazelcast-cpp-client-4.2.0/hazelcast/src/hazelcast/util/util.cpp: In function 'int hazelcast::util::strerror_s(int, char*, size_t, const char*)':
/data/hazelcast-cpp-client-4.2.0/hazelcast/src/hazelcast/util/util.cpp:417:40: error: invalid conversion from 'int' to 'char*' [-fpermissive]
  417 |             char *errStr = ::strerror_r(errnum, strerrbuf + numChars, buflen - numChars);
      |                            ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                        |
      |                                        int
```
[
musl's strerror_r](http://git.musl-libc.org/cgit/musl/tree/src/string/strerror_r.c) returns `int`, not `char*`.

[See](https://linux.die.net/man/3/strerror_r) for more info.

The fix is just to remove `hazelcast::util::strerror_s`, it's not used anywhere other than tests.

--- a/hazelcast/include/hazelcast/util/Util.h
+++ b/hazelcast/include/hazelcast/util/Util.h
@@ -64,12 +64,6 @@ namespace hazelcast {
          */
         HAZELCAST_API int64_t current_time_nanos();
 
-        /**
-         * @return 0 if error string could be obtained, non-zero otherwise
-         */
-        HAZELCAST_API int
-        strerror_s(int errnum, char *strerrbuf, size_t buflen, const char *msg_prefix = (const char *) NULL);
-
         int32_t HAZELCAST_API get_available_core_count();
 
         template <typename T>
@@ -87,19 +81,6 @@ namespace hazelcast {
              * @return the the formatted time string. Format is "%Y-%m-%d %H:%M:%S.%f".
              */
             static std::string time_to_string(std::chrono::steady_clock::time_point t);
-
-            /**
-             * Tokenizes a version string and returns the tokens with the following grouping:
-             * (1) major version, eg "3"
-             * (2) minor version, eg "8"
-             * (3) patch version prefixed with ".", if exists, otherwise {@code null} (eg ".0")
-             * (4) patch version, eg "0"
-             * (5) 1st -qualifier, if exists
-             * (6) -SNAPSHOT qualifier, if exists
-             * @param version
-             * @return
-             */
-            static std::vector<std::string> tokenize_version_string(const std::string &version);
         };
 
         class HAZELCAST_API Int64Util {

--- a/hazelcast/src/hazelcast/util/util.cpp
+++ b/hazelcast/src/hazelcast/util/util.cpp
@@ -385,44 +385,6 @@ namespace hazelcast {
                     std::chrono::system_clock::now().time_since_epoch()).count();
         }
 
-        int strerror_s(int errnum, char *strerrbuf, size_t buflen, const char *msg_prefix) {
-            int numChars = 0;
-            if ((const char *) NULL != msg_prefix) {
-                numChars = util::hz_snprintf(strerrbuf, buflen, "%s ", msg_prefix);
-                if (numChars < 0) {
-                    return numChars;
-                }
-
-                if (numChars >= (int) buflen - 1) {
-                    return 0;
-                }
-            }
-
-#if  defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
-            if (!FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM,
-                  NULL,
-                  errnum,
-                  0,
-                  (LPTSTR)(strerrbuf + numChars),
-                  buflen - numChars,
-                  NULL)) {
-                return -1;
-            }
-            return 0;
-#elif defined(__llvm__) && !_GNU_SOURCE
-            /* XSI-compliant */
-            return ::strerror_r(errnum, strerrbuf + numChars, buflen - numChars);
-#else
-            /* GNU-specific */
-            char *errStr = ::strerror_r(errnum, strerrbuf + numChars, buflen - numChars);
-            int result = util::hz_snprintf(strerrbuf + numChars, buflen - numChars, "%s", errStr);
-            if (result < 0) {
-                return result;
-            }
-            return 0;
-#endif
-        }
-
         int32_t get_available_core_count() {
             return (int32_t) std::thread::hardware_concurrency();
         }
@@ -450,13 +412,6 @@ namespace hazelcast {
             oss << '.' << std::setfill('0') << std::setw(3) << msecs;
 
             return oss.str();
-        }
-
-        std::vector<std::string> StringUtil::tokenize_version_string(const std::string &version) {
-            // passing -1 as the submatch index parameter performs splitting
-            std::vector<std::string> result;
-            boost::split(result, version, boost::is_any_of("."));
-            return result;
         }
 
         int Int64Util::number_of_leading_zeros(int64_t i) {

--- a/hazelcast/test/src/HazelcastTests2.cpp
+++ b/hazelcast/test/src/HazelcastTests2.cpp
@@ -143,27 +143,6 @@ namespace hazelcast {
             protected:
             };
 
-            TEST_F (ClientUtilTest, testStrError) {
-#if  defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
-                                                                                                                                        int error = WSAEINPROGRESS;
-                std::string expectedErrorString("A blocking operation is currently executing.\r\n");
-#else
-                int error = EINPROGRESS;
-                std::string expectedErrorString("Operation now in progress");
-#endif
-
-                char msg[100];
-                const std::string prefix = "testStrError prefix message";
-                ASSERT_EQ(0, hazelcast::util::strerror_s(error, msg, 100, prefix.c_str()));
-                ASSERT_STREQ((prefix + " " + expectedErrorString).c_str(), msg);
-
-                ASSERT_EQ(0, hazelcast::util::strerror_s(error, msg, prefix.length() + 1, prefix.c_str()));
-                ASSERT_STREQ(prefix.c_str(), msg);
-
-                ASSERT_EQ(0, hazelcast::util::strerror_s(error, msg, 100));
-                ASSERT_STREQ(expectedErrorString.c_str(), msg);
-            }
-
             TEST_F (ClientUtilTest, testAvailableCoreCount) {
                 ASSERT_GT(hazelcast::util::get_available_core_count(), 0);
             }

